### ISSUE TITLE
CA1849 triggers when using nameof of a Task property in an async method

### ISSUE THREAD

**Issue by GiottoVerducci** [dotnet/roslyn-analyzers#6993]
### Analyzer

**Diagnostic ID**: [CA1849](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1849): `Call async methods when in an async method`

### Analyzer source

**NuGet Package**: microsoft.entityframeworkcore.analyzers 
**Version**: 7.0.12 (latest) 

### Describe the bug

The following expression triggers a warning when used in an async method

`nameof(Task<object>.Result)`

For instance:

```
        public async Task<string> Foo()
        {
            await Task.CompletedTask; // do async stuff
            return nameof(Task<object>.Result); // this should return "Result"
        }

```

### Expected behavior

Since it's a nameof(), it shouldn't trigger the warning.

### Actual behavior

It triggers a warning: 
Warning	CA1849	'Task<TResult>.Result' synchronously blocks. Use await instead.
